scipy/sparse/dok.py

```diff
@@ -414,6 +411,9 @@ def __itruediv__(self, other):
         else:
             return NotImplemented
 
+    def __reduce__(self):
+        return dict.__reduce__(self)
```

Comment:
``test_pickle`` that I added fails on python3 for me on 2 ubuntu machines. I just tested and it works in python 2 thought. From debugging I found that it goes to ``spmatrix.__reduce__`` for whatever reason, and that returns an empty dict. I am still not completely sure that this is the right fix though because of all the nuances of subclassing a dict.